akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/BehaviorSetup.scala

```diff
@@ -173,6 +176,110 @@ private[akka] final class BehaviorSetup[C, E, S](
     }
   }
 
+  // The retention process for SnapshotCountRetentionCriteria looks like this:
+  // 1. Save snapshot after persisting events when shouldSnapshotAfterPersist returned SnapshotWithRetention.
+  // 2. Delete events (when deleteEventsOnSnapshot=true), runs in background.
+  // 3. Delete snapshots (when isOnlyOneSnapshot=false), runs in background.
+
+  def isRetentionInProgress(): Boolean =
+    retentionInProgress
+
+  def retentionProgressSaveSnapshot(sequenceNr: Long): Unit = {
```

Comment:
This sounds like it does save snapshot so I started wondering why it doesn't do anything, `retentionProgressWillSaveSnapshot()` or something like that would be more clear.

Reply:
renamed to *Started *Ended